### PR TITLE
Hemant - solution

### PR DESCRIPTION
**Pull request by hemant-saikia**
First submission. Haven't updated the time and space complexity. Will do shortly.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  },
+  {
+    name: "Life of Pi",
+    license: [2008,2014]
+  },
+  {
+    name: "Another Movie",
+    license: [2011,2013]
+  }
+];
```
Comment (ankushdharkar) [inline app.js]: Let's put this in a separate file

Also, let's use `const`

File: app.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+
+
+var frequency = {};
```
Comment (ankushdharkar) [inline app.js]: Avoid global variables

File: app.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+function main(arr)
+{
+	countYearFrequency(arr)
```
Comment (ankushdharkar) [inline app.js]: Avoid mutating the given array

If someone else on the team, or some other code is also using the array to compute some data, they have lost the original array. Try not to modify inputs, until absolutely necessary

File: app.js [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+	arr.forEach(updateYearFrequency);     
+}
+
+function updateYearFrequency(item)
+{
+	startOfRange = item.license[0]
+	endOfRange = item.license[1]
+	
+	var i;
+	for(i = startOfRange; i<=endOfRange; i++ )
```
Comment (ankushdharkar) [inline app.js]: `for(let i= ...)` 
No need to declare `i` above the loop

Read up on scope of `var`

and differences between `var`, `const`, `let`

File: app.js [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+	{
+		if (i in frequency)
+			frequency[i]++
+
+		else
+			frequency[i] = 1
```
Comment (ankushdharkar) [inline app.js]: Can be reduced to one-liner using array prototype functions

File: app.js [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+	}
+}
+
+function getMaxYear(frequency)
+{
+	year = []
```
Comment (ankushdharkar) [inline app.js]: var, let, const?

File: app.js [hunk 7]
```diff
@@ -0,0 +1,74 @@
...
+  		if (frequency[key] == max)
+  		{
+  			year.push(key)
+  		}
+  		else if (frequency[key] > max)
+  		{
+  			max = frequency[key]
+  			year = [key]
+  		}
+	}
```
Comment (ankushdharkar) [inline app.js]: Reduce code by using `Object.entries` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/entries

**Review by ankushdharkar — CHANGES_REQUESTED** [review]

File: app.js (hunk 6 above)
Comment (hemant-saikia) [reply app.js]: Added let

File: app.js (hunk 3 above)
Comment (hemant-saikia) [reply app.js]: I don't think I am mutating the array. Am I missing something ?

File: app.js (hunk 2 above)
Comment (hemant-saikia) [reply app.js]: Okay. noted.

File: app.js (hunk 1 above)
Comment (hemant-saikia) [reply app.js]: Using const. How do I import from a separate file ?

File: app.js (hunk 7 above)
Comment (hemant-saikia) [reply app.js]: Reduced a lot of code.

**Pull request closed without merging**